### PR TITLE
Instantiate2 module install

### PR DESCRIPTION
**Pull request by Buckram123**
Demo of Instantiate2 approach discussed in ModuleInstall notion
If we commit to this approach, TBD:
- [ ] Update instantiation of manager and proxy via instantiate2, which will help removing caching modules(and AccountFactory as "admin") to install modules and updating admins during instantiation

### Checklist

- [ ] CI is green.
- [ ] Changelog updated.

### CODE REVIEW

**Review by CyberHoward — COMMENTED** [review]
Some initial comments, need to review more

File: framework/contracts/account/manager/src/contract.rs [hunk 1]
```diff
@@ -108,17 +108,14 @@ pub fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg) -> M
...
-                    update_internal_config(deps, info, config)
+                    update_internal_config(deps, env, info, config)
                 }
                 ExecuteMsg::SetOwner { owner } => set_owner(deps, env, info, owner),
 
                 ExecuteMsg::InstallModules { modules } => install_modules(deps, info, env, modules),
                 ExecuteMsg::UninstallModule { module_id } => {
                     uninstall_module(deps, info, module_id)
                 }
-                ExecuteMsg::RegisterModules { modules } => {
```
[the remark this answers is not in the document]
Comment (Buckram123) [reply framework/contracts/account/manager/src/contract.rs:119]: Same as apps, why would we need this endpoint for standalones?

File: framework/contracts/account/manager/src/commands.rs [hunk 2]
```diff
@@ -89,114 +95,145 @@ pub fn update_module_addresses(
...
+                    &canonical_module_factory,
+                    &salt,
+                )?;
+                let module_address = deps.api.addr_humanize(&module_address)?;
+                to_add.push((module.info.id(), module_address.to_string()));
+                install_context.push((module.clone(), Some(module_address)));
+
+                Some(init_msg.unwrap())
+            }
+            // TODO: do we want to support installing any other type of module here?
```
Comment (CyberHoward) [inline framework/contracts/account/manager/src/commands.rs:180]: See my earlier comment. Does this mean standalone is not supported?

File: framework/contracts/account/manager/src/commands.rs [hunk 3]
```diff
@@ -89,114 +95,145 @@ pub fn update_module_addresses(
...
-        installed_modules.push(module.id_with_version());
+        installed_modules.push(module.info.id_with_version());
+
+        let init_msg_salt = match &module.reference {
+            ModuleReference::Adapter(module_address) => {
+                to_add.push((module.info.id(), module_address.to_string()));
+                install_context.push((module.clone(), None));
+                None
+            }
+            ModuleReference::App(code_id) | ModuleReference::Standalone(code_id) => {
```
Comment (Buckram123) [inline framework/contracts/account/manager/src/commands.rs:167]: Standalones supported! @CyberHoward

Comment (CyberHoward) [reply framework/contracts/account/manager/src/commands.rs:167]: wow, completely missed that line

File: framework/contracts/account/manager/tests/apps.rs [hunk 4]
```diff
@@ -72,10 +72,18 @@ fn account_install_app() -> AResult {
...
 
-    let app = MockApp::new_test(chain);
+    let app = MockApp::new_test(chain.clone());
     app.deploy(APP_VERSION.parse().unwrap(), DeployStrategy::Try)?;
     let app_addr = account.install_app(app, &MockInitMsg, None)?;
     let module_addr = account.manager.module_info(APP_ID)?.unwrap().address;
+
     assert_that!(app_addr).is_equal_to(module_addr);
+
+    let contract_info = chain
```
Comment (CyberHoward) [inline framework/contracts/account/manager/tests/apps.rs:82]: Was this used for debugging?

**Review by CyberHoward — COMMENTED** [review]
Can we keep `ManagerModuleInstall` as `ModuleInstallConfig` and rename the current `ModuleInstallConfig` to `FactoryModuleInstallConfig`?

Because we won't be exposing the factory one so any devs should only care about the normal one so we can keep it simple.

**Comment by CyberHoward** [comment]
> "Update instantiation of manager and proxy via instantiate2, which will help removing caching modules(and AccountFactory as "admin") to install modules and updating admins during instantiation" 

Let's merge this as it is and create a separate PR for Manager and Proxy instantiate2 changes

File: framework/packages/abstract-app/src/endpoints/instantiate.rs [hunk 5]
```diff
@@ -54,6 +54,9 @@ impl<
             address: deps.api.addr_validate(&version_control_address)?,
         };
 
+        // TODO: Would be nice to remove context
+        // Issue: We can't pass easily AccountBase with BaseInstantiateMsg(right now)
```
Comment (CyberHoward) [inline framework/packages/abstract-app/src/endpoints/instantiate.rs:58]: https://linear.app/abstract-sdk/issue/ABS-229/remove-requirement-for-base-message-in-app-instantiate-message

File: framework/contracts/account/manager/tests/apps.rs (hunk 4 above)
Comment (Buckram123) [reply framework/contracts/account/manager/tests/apps.rs]: Yep

**Review by AndrewPochapsky — APPROVED** [review]

**Pull request merged**